app/views/taxon_frameworks/_form.html.haml

```diff
@@ -0,0 +1,92 @@
+    .source input.text {width: 300px;}
+    
+- content_for :extrajs do
+  :javascript
+    $( document ).ready( function() {
+      
+      $( "#taxon_name" ).taxonAutocomplete( { idEl: $( "#taxon_framework_taxon_id" ) } );
+      
+      var is_admin = #{current_user ? (current_user.is_admin? ? true : false ) : false };
+      if(is_admin == false){
+        $( "#covers_section" ).hide();
+      }
+      
+      var had_references = false;
+      function show_confirm() {
+        if( had_references == true ){
+          $( 'input[type=submit]' ).
+            attr( "data-confirm", "It looks like you might have changed the Source or the Rank level. If so saving will destroy all associated Taxon Framework Relationships. Are you sure?" );
+        }
+      }
+      
+      if( $( "#taxon_framework_rank_level" ).val() != "" && is_admin == true) {
+        $( "#covers" ).prop('checked', true);
+        $( "#covers_details" ).show();
+        if( $( ".source_nested_form_fields" ).find( ".existing" ).val() != "" ) {
+          $( "#add_source" ).prop('checked', true);
+          $( "#add_source_details" ).show();
+          had_references = true;
+          $( "#taxon_framework_rank_level" ).change( show_confirm );
+          $( ".source_nested_form_fields" ).change( show_confirm );
+          $( "#taxon_name" ).attr( "disabled", true ).removeClass( "text" ).
+            addClass("form-control-plaintext");
+        }
+      }
+
+      $( "#covers" ).click( function() {
+        if( this.checked ){
+          $( "#covers_details" ).show( 500 );
+        }else{
+          if( $( "#add_source" ).is( ":checked" ) ){
+            $( "#add_source" ).prop( "checked", false);
+            $( ".source_nested_form_fields" ).find( ".existing" ).chooser( "clear" );
+            show_confirm();
+            $( "#add_source_details" ).hide( 500 );
+          }
+          $( "#covers_details" ).hide( 500 );
+          $( "#taxon_framework_rank_level" ).val( "" );
+          show_confirm();
+          $( "#taxon_framework_complete" ).prop( "checked", false );
+        }
+      });
+      
+      $( "#add_source" ).click( function() {
+        if( this.checked ){
+          $( "#add_source_details" ).show( 500 );
+        }else{
+          $( ".source_nested_form_fields" ).find( ".existing" ).chooser( "clear" );
+          show_confirm();
+          $( "#add_source_details" ).hide( 500 );
+        }
+      } );
+      
+    } );
+= form_for @taxon_framework, builder: BootstrapFormBuilder do |f|
+  .form-group
+    = f.form_field :taxon do
+      = text_field_tag :taxon_name, "", class: "form-control", placeholder: t( :start_typing_taxon_name )
+      = f.hidden_field :taxon_id, value: f.object.taxon_id || params[:taxon_id]
+
+  .form-group
+    = f.text_area :description, class: "form-control"
+  
+  .form-group#covers_section
+    = label_tag 'Make it a cover taxa downstream'
+    = check_box_tag :covers
+    
+    #covers_details{ :style=>"display:none;" }
```

Comment:
If you're going to use `=>` put spaces around it, but this should be style: "display: none". Actually, there just shouldn't be an embedded style attribute and this should be in the stylesheet, or you can use the `hidden` CSS class that comes with bootstrap.

Reply:
done